File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//    品牌服务 抽取涉及服务的方法 集中
app.service("brandService",function($http){
    this.findAll=function(){
        return $http.get("../brand/findAll.do");
    };

    this.findPage = function(page,size){
        return $http.get("../brand/findPage.do?page="+page+"&size="+size);
    };

    this.findOne = function(id) {
        return $http.get("../brand/findOne.do?id="+id);
    };

    this.add = function(entity){
        return $http.post("../brand/add.do",entity);
    };

    this.update = function(entity){
        return $http.post("../brand/update.do",entity);
    };

    this.dele = function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    };

    this.search = function (page,size,searchEntity) {
        return $http.post("../brand/search.do?page="+page+"&size="+size,searchEntity);
    }
});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    public List<TbBrand> findAll();

    /**
     * 分页
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    public void add(TbBrand tbBrand);

    TbBrand findOne(Long id);

    void update(TbBrand tbBrand);

    void delete(Long[] ids);

    public PageResult findPage(TbBrand tbBrand,int pageNum, int pageSize);

    /**
     * 返回下拉列表
     * @return
     */
    public List<Map> selectOptionList();
}
